### PR TITLE
doc: add Snippet (Name User Code POJO) in nested element and update Sales Documents

### PR DESCRIPTION
**Pull request by Tiity**

### CODE REVIEW

File: _config.yml [hunk 1]
```diff
@@ -99,3 +99,4 @@ format:
     periods: '<a href="/docs/1.0/api/#cash_flow_overtime_periods" alt=""/>Periods</a>'
     top_customers: 'array[<a href="/docs/1.0/api/#top_customer" alt=""/>Top customer</a>]'
     top_invoiced: 'array[<a href="/docs/1.0/api/#top_invoiced" alt=""/>Top invoiced</a>]'
+    snippet_customer: '<a href="/docs/1.0/api/#snippet" title="">Snippet</a>&lt;<a href="/docs/1.0/api/#customer" title="">Customer</a>&gt;'
```
Comment (fChalmeau) [inline _config.yml:102]: Actually, that type should be more generic and just 'snippet'.

We will have more snippet in the futur, like Invoice snippet, etc... they should all be linked to the 'snippet' description of the documentation.

File: _includes/1.0/api/creditmemo.html [hunk 2]
```diff
@@ -55,12 +55,26 @@ <h3 id="creditmemojsonformat">JSON Format</h3>
...
-                <td>customer_id</td>
+                <td>customer_id <span class="deprecated">DEPRECATED</span></td>
                 <td>{{ site.format.type.integer }}</td>
                 <td>yes</td>
                 <td>no</td>
                 <td>Id of the <a href="/docs/1.0/api/#customer" title="">Customer.</a></td>
             </tr>
+            <tr>
+                <td>customer</td>
+                <td>{{ site.format.type.snippet_customer }}</td>
```
Comment (fChalmeau) [inline _includes/1.0/api/creditmemo.html:66]: Related to the previous comment, here we will have `site.format.type.snippet`, linked to the common "snippet" section of the documentation.

File: _includes/1.0/api/creditmemo.html [hunk 3]
```diff
@@ -55,12 +55,26 @@ <h3 id="creditmemojsonformat">JSON Format</h3>
...
+                <td>yes</td>
+                <td>no</td>
+                <td>
+                    The <a href="/docs/1.0/api/#snippet" title="">Snippet</a> of the credit memo
```
Comment (fChalmeau) [inline _includes/1.0/api/creditmemo.html:70]: Then here the hlink to "snippet" type is not needed, but we keep the one to the "customer" section.

So, the 'snippet' type is generic, then the mapping of each particular snippet is explained in the field description.

File: _includes/1.0/api/nestedelements.html [hunk 4]
```diff
@@ -244,4 +243,37 @@ <h3>JSON Format</h3>
         <h3>JSON Example</h3>
         {% highlight bash %}{% include 1.0/api/json/orderline.json %}{% endhighlight %}
     </div>
+
+    <div id="snippet" class="group">
+        <h2>Snippet</h2>
+    	<p>A Snippet is a generic type that encapsulates an object and aggregates common information.</p>
```
Comment (fChalmeau) [inline _includes/1.0/api/nestedelements.html:249]: Nice description 👍

**Review by fChalmeau — CHANGES_REQUESTED** [review]
Some changes to be done to keep the 'snippet' type generic.

**Comment by Tiity** [comment]
@fChalmeau fix with 9940e68

**Pull request merged**